packages/core/src/storages/request_queue_v2.ts

```diff
@@ -204,21 +210,28 @@ export class RequestQueue extends RequestProvider {
     }
 
     private async _listHeadAndLock(): Promise<void> {
+        // TODO Kuba: This is a little bit confusing, let's rename to something descriptive.
         const forefront = await this.hasPendingForefrontRequests();
 
+        // TODO Kuba: The list and lock is ATOMIC operation, so we can be sure that we are the only ones reading these requests for requestLockSecs nobody else can read them using listAndLockHead.
         const headData = await this.client.listAndLockHead({
             limit: Math.min(forefront ? this.assumedForefrontCount : 25, 25),
             lockSecs: this.requestLockSecs,
         });
 
         const headIdBuffer = [];
 
+        // TODO Kuba: This is a bit of hack this should not happen. We need to investigate why.
```

Comment:
I suppose you mean unlocking the forefront requests? I guess @barjin could provide some context here?

Reply:
Anyway, this is the PR that added it https://github.com/apify/crawlee/pull/2689/files

Reply:
from meeting:
this fix only if we implement ensure head and lock